stream.go

```diff
@@ -0,0 +1,355 @@
+	Before        *time.Time `url:"before,omitempty"`
+	Creator       string     `url:"creator,omitempty"`
+	IncludeCounts bool       `url:"include_counts,omitempty"`
+	Search        string     `url:"search,omitempty"`
+	Limit         int        `url:"limit,omitempty"`
+	Asc           bool       `url:"asc,omitempty"`
+	Status        string     `url:"status,omitempty"`
+}
+
+// StreamSignedURLOptions represent parameters used when creating a signed URL.
+type StreamSignedURLOptions struct {
+	ID           string            `json:"id,omitempty"`
+	PEM          string            `json:"pem,omitempty"`
+	EXP          int               `json:"exp,omitempty"`
+	NBF          int               `json:"nbf,omitempty"`
+	Downloadable bool              `json:"downloadable,omitempty"`
+	AccessRules  StreamAccessRules `json:"accessRules,omitempty"`
+}
+
+// StreamVideoResponse represents an API response of a stream video.
+type StreamVideoResponse struct {
+	Response
+	Result StreamVideo `json:"result,omitempty"`
+}
+
+// StreamVideoCreateResponse represents an API response of creating a stream video.
+type StreamVideoCreateResponse struct {
+	Response
+	Result StreamVideoCreate `json:"result,omitempty"`
+}
+
+// StreamListResponse represents the API response from a StreamListRequest.
+type StreamListResponse struct {
+	Response
+	Result []StreamVideo `json:"result,omitempty"`
+	Total  string        `json:"total,omitempty"`
+	Range  string        `json:"range,omitempty"`
+}
+
+// StreamSignedURLResponse represents an API response for a signed URL.
+type StreamSignedURLResponse struct {
+	Response
+	Result struct {
+		Token string `json:"token,omitempty"`
+	}
+}
+
+// StreamAccessRules represents the accessRules when creating a signed URL
+//I feel like should be implemented somewhere already but couldn't find it
+// accessRules under https://api.cloudflare.com/#stream-videos-create-a-signed-url-token-for-a-video
+type StreamAccessRules []map[string]interface{}
+
+// StreamUploadFromURL send a video URL to it will be downloaded and made available on Stream.
+//
+// API Reference: https://api.cloudflare.com/#stream-videos-upload-a-video-from-a-url
+func (api *API) StreamUploadFromURL(ctx context.Context, accountID string, options StreamUploadFromURLOptions) (StreamVideo, error) {
+	if options.URL == "" {
+		return StreamVideo{}, ErrMissingUploadURL
+	}
+	uri := fmt.Sprintf("/accounts/%s/stream/copy", accountID)
+
+	res, err := api.makeRequestContext(ctx, http.MethodPost, uri, options)
+	if err != nil {
+		return StreamVideo{}, err
+	}
+
+	var r StreamVideoResponse
+	if err := json.Unmarshal(res, &r); err != nil {
+		return StreamVideo{}, err
+	}
+	return r.Result, nil
+}
+
+// StreamUploadVideoFile uploads a video from a path to the file.
+//
+// API Reference: https://api.cloudflare.com/#stream-videos-upload-a-video-using-a-single-http-request
+func (api *API) StreamUploadVideoFile(ctx context.Context, accountID string, filePath string) (StreamVideo, error) {
```

Comment:
similar to https://github.com/cloudflare/cloudflare-go/pull/959/files#r909157172. check out https://github.com/cloudflare/cloudflare-go/blob/master/tunnel.go for some examples of passing in multiple parameters to a method.

Reply:
This seems excessive for methods that don't already require options like

https://github.com/cloudflare/cloudflare-go/pull/959/files#diff-4f588cddf63cad8c5e4fc9685c5b200fb4daacf14be2bece09a8296d4e266d90R283

https://github.com/cloudflare/cloudflare-go/pull/959/files#diff-4f588cddf63cad8c5e4fc9685c5b200fb4daacf14be2bece09a8296d4e266d90R283

https://github.com/cloudflare/cloudflare-go/pull/959/files#diff-4f588cddf63cad8c5e4fc9685c5b200fb4daacf14be2bece09a8296d4e266d90R283

Should AccountID and VideoID both be put into the options, or should the options setting be ignored for those that only require accountID and VideoID?

Reply:
yes, all parameters to go into the struct. it aligns with where we are taking the next generation of this client. see https://github.com/cloudflare/cloudflare-go/blob/master/docs/experimental.md for more details.